src/models/tournament/services/tournaments.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CreateTournamentDto, TournamentDto, TournamentsList, TournamentsQuery } from '../dto/tournaments.dto';
+import { Tournament } from '../entities/tournament.entity';
+@Injectable()
+export class TournamentsService {
+	private readonly logger = new Logger(TournamentsService.name);
+
+	constructor(
+		@InjectRepository(Tournament)
+		private tournaments: Repository<Tournament>,
+	) { }
+
+	generateSearchQuery(query: TournamentsQuery) {
+		const search = {};
+		const { finished } = query;
+		if (finished == true || finished == false) {
+			search['finished'] = finished
+		}
+
+		return search;
+	}
+
+	async getAll(query?: TournamentsQuery): Promise<TournamentsList> {
+		const search = this.generateSearchQuery(query);
+		try {
+			const result = await this.tournaments.find({
+				where: search,
+				order: { 'id': 1 }
+			});
+
+			return {
+				tournaments: result ?? [],
+			};
+		} catch (error) {
+			this.logger.error(error);
+			throw new Error('Could not get tournaments. ' + error);
+		}
+	}
+
+	async getEntireTournamentById(id: number): Promise<TournamentDto> {
+		if (!Number.isSafeInteger(id)) {
+			throw new Error(`id must be an integer, but was '${id}'`);
+		}
+
+		return await this.tournaments
+			.findOneOrFail({
+				where: { id },
+				relations: {
```

Comment:
Yeah this is where having flat entities/JSON blobs is really nice, deeply nested relations in typeorm is never fun....

Reply:
Indeed, it requires a bit more boilerplate.

I'll resolve this under the assumption that this is not an inquiry for change